server/controllers/finance/vouchers.js

```diff
@@ -36,34 +36,49 @@ exports.create = create;
 /** Get the voucher report */
 exports.report = report;
 
+/** Get the voucher receipt */
+exports.receipt = receipt;
+
 /**
 * GET /vouchers
 *
 * @method list
 */
 function list(req, res, next) {
-  var query =
+  'use strict';
+
+  let dateConditon = null;
+  let query =
     `SELECT BUID(v.uuid) as uuid, v.date, v.project_id, v.currency_id, v.amount,
       v.description, BUID(vi.document_uuid) as document_uuid,
       v.user_id, BUID(vi.uuid) AS voucher_item_uuid,
-      vi.account_id, vi.debit, vi.credit,
+      vi.account_id, vi.debit, vi.credit, v.origin_id,
+      a.number AS account_number, a.label AS account_label,
+      CONCAT(u.first, ' - ', u.last) AS user,
       CONCAT(p.abbr, v.reference) AS reference
     FROM voucher v
     JOIN voucher_item vi ON vi.voucher_uuid = v.uuid
-    JOIN project p ON p.id = v.project_id `;
+    JOIN project p ON p.id = v.project_id
+    JOIN user u ON u.id = v.user_id
+    JOIN account a ON a.id = vi.account_id `;
 
   // convert binary params if they exist
   if (req.query.document_uuid) {
     req.query.document_uuid = db.bid(req.query.document_uuid);
   }
 
+  /**
+   * the date conditions string
+   */
+  if (req.query.dateFrom && req.query.dateTo) {
+    dateConditon = 'DATE(v.date) BETWEEN DATE(?) AND DATE(?)';
+  }
+
   // format query parameters appropriately
-  var builder = util.queryCondition(query, req.query);
+  let builder = util.queryCondition(query, req.query, null, dateConditon);
```

Comment:
Hmm...  this function's API is getting very big and confusing.  We should refactor it soon.